tests/core/_orchestrator/_dispatcher/test_task_function_wrapper.py

```diff
@@ -0,0 +1,138 @@
+from taipy.core.task.task import Task
+
+
+def _create_task(function, nb_outputs=1):
+    output_dn_config_id = "".join(random.choice(string.ascii_lowercase) for _ in range(10))
+    dn_input_configs = [
+        Config.configure_data_node("input1", "pickle", Scope.SCENARIO, default_data=21),
+        Config.configure_data_node("input2", "pickle", Scope.SCENARIO, default_data=2),
+    ]
+    dn_output_configs = [
+        Config.configure_data_node(f"{output_dn_config_id}_output{i}", "pickle", Scope.SCENARIO, default_data=0)
+        for i in range(nb_outputs)
+    ]
+    input_dn = _DataManager._bulk_get_or_create(dn_input_configs).values()
+    output_dn = _DataManager._bulk_get_or_create(dn_output_configs).values()
+    return Task(
+        output_dn_config_id,
+        {},
+        function=function,
+        input=input_dn,
+        output=output_dn,
+    )
+
+
+def multiply(nb1: float, nb2: float):
+    sleep(0.1)
+    return nb1 * nb2
+
+
+def test_execute_task_that_return_multiple_outputs():
+    def return_2tuple(nb1, nb2):
+        return multiply(nb1, nb2), multiply(nb1, nb2) / 2
+
+    def return_list(nb1, nb2):
+        return [multiply(nb1, nb2), multiply(nb1, nb2) / 2]
+
+    with_tuple = _create_task(return_2tuple, 2)
+    with_list = _create_task(return_list, 2)
+    _TaskFunctionWrapper("job_id_tuple", with_tuple).execute()
+    _TaskFunctionWrapper("job_id_list", with_list).execute()
+
+    assert (
+        with_tuple.output[f"{with_tuple.config_id}_output0"].read()
+        == with_list.output[f"{with_list.config_id}_output0"].read()
+        == 42
+    )
+    assert (
+        with_tuple.output[f"{with_tuple.config_id}_output1"].read()
+        == with_list.output[f"{with_list.config_id}_output1"].read()
+        == 21
+    )
+
+
+def test_execute_task_that_returns_single_iterable_output():
+    def return_2tuple(nb1, nb2):
+        return multiply(nb1, nb2), multiply(nb1, nb2) / 2
+
+    def return_list(nb1, nb2):
+        return [multiply(nb1, nb2), multiply(nb1, nb2) / 2]
+
+    task_with_tuple = _create_task(return_2tuple, 1)
+    task_with_list = _create_task(return_list, 1)
+    _TaskFunctionWrapper("job_id_tuple", task_with_tuple).execute()
+    _TaskFunctionWrapper("job_id_list", task_with_list).execute()
+
+    assert task_with_tuple.output[f"{task_with_tuple.config_id}_output0"].read() == (42, 21)
+    assert len(_OrchestratorFactory._dispatcher._dispatched_processes) == 0
+    assert task_with_list.output[f"{task_with_list.config_id}_output0"].read() == [42, 21]
+    assert len(_OrchestratorFactory._dispatcher._dispatched_processes) == 0
+
+
+def test_data_node_not_written_due_to_wrong_result_nb():
+    def fct_2_outputs():
+        return lambda nb1, nb2: (multiply(nb1, nb2), multiply(nb1, nb2) / 2)
+
+    task_expecting_3_outputs = _create_task(fct_2_outputs, 3)
+
+    res = _TaskFunctionWrapper("job_id", task_expecting_3_outputs).execute()
+
+    assert len(res) == 1
+    assert isinstance(res[0], Exception)
```

Comment:
```suggestion
    exceptions = _TaskFunctionWrapper("job_id", task_expecting_3_outputs).execute()

    assert len(exceptions) == 1
    assert isinstance(exceptions[0], Exception)
```
I would name this `exceptions` for better readability of the test. It's not important though.